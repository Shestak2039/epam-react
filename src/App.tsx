import React from 'react';

import Header from './components/Header/Header';
import Wrapper from './components/Wrapper/Wrapper';
import Main from './components/Main/Main';
import SortingFilms from './components/SortingFilms/SortingFilms';

const App: React.FunctionComponent = () => (
    <Wrapper>
        <Header />
        <Main>
            <SortingFilms />
        </Main>
    </Wrapper>
);

export default App;
