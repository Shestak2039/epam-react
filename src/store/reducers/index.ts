import { combineReducers } from 'redux';

import filmsReducer from './films-reducer';

export default combineReducers({
    filmsState: filmsReducer
});
