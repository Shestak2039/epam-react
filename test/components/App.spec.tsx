import React from 'react';
import { render } from '@testing-library/react';

import App from '../../src/App';
import { Provider } from 'react-redux';
import store from '../../src/store/store';


describe('Footer', () => {
    test('renders Footer snapshot', () => {
        const { asFragment } = render(<Provider store={store}><App /></Provider>);

        expect(asFragment()).toMatchSnapshot();
    });
});
