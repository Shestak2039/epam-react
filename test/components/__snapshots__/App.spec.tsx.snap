// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer snapshot 1`] = `
<DocumentFragment>
  <div
    class="background"
  >
    <div
      class="wrapper"
    >
      <header
        class="header"
      >
        <div
          class="header-search"
        >
          <div
            class="header-top"
          >
            <button
              class="button default"
            >
              + ADD MOVIE
            </button>
          </div>
          <nav
            class="header-nav"
          >
            <div
              class="header-nav-title"
            >
              FIND YOUR MOVIE
            </div>
            <div
              class="search-bar"
            >
              <div
                class="search"
              >
                <input
                  class="search-input"
                  id="search-input"
                  name="search-input"
                  placeholder="What do you want to watch?"
                  value=""
                />
              </div>
              <a
                href="/search/"
              >
                <button
                  class="button primary"
                >
                  SEARCH
                </button>
              </a>
            </div>
          </nav>
        </div>
      </header>
      <main
        class="main"
      >
        <div
          class="sorting-bar"
        >
          <ul
            class="genres"
          >
            <li
              class="genre selected"
            >
              ALL
            </li>
            <li
              class="genre"
            >
              DOCUMENTARY
            </li>
            <li
              class="genre"
            >
              DRAMA
            </li>
            <li
              class="genre"
            >
              HORROR
            </li>
            <li
              class="genre"
            >
              THRILLER
            </li>
          </ul>
          <div
            class="filter"
          >
            <div
              class="filter-name"
            >
              SORT BY
            </div>
            <select
              class="filter-select"
            >
              <option
                class="filter-select-option"
                selected=""
                value="1"
              >
                RATING
              </option>
              <option
                class="filter-select-option"
                value="2"
              >
                YEAR
              </option>
            </select>
          </div>
        </div>
        <div
          class="films-wrapper"
        >
          <div
            class="count-movies"
          >
            0 movies found
          </div>
          <div
            class="films"
          />
        </div>
      </main>
      <footer
        class="footer"
      />
    </div>
  </div>
</DocumentFragment>
`;
